src/Filesystem/Mime.php

```diff
@@ -268,17 +269,31 @@ public static function toExtension(string $mime = null): string|false
 	/**
 	 * Returns all available extensions for a given MIME type
 	 */
-	public static function toExtensions(string $mime = null): array
+	public static function toExtensions(string $mime = null, bool $matchWildcards = false): array
 	{
 		$extensions = [];
+		$testMime = fn (string $v) => static::matches($v, $mime);
 
 		foreach (static::$types as $key => $value) {
-			if (is_array($value) === true && in_array($mime, $value) === true) {
-				$extensions[] = $key;
+			$isArray = is_array($value) === true;
+
+			if ($matchWildcards === true) {
+				if ($isArray === true && A::some($value, $testMime)) {
+					$extensions[] = $key;
+				}
+
+				if ($isArray === false && $testMime($value) === true) {
+					$extensions[] = $key;
+				}
+
 				continue;
 			}
 
-			if ($value === $mime) {
+			if ($isArray === true && in_array($mime, $value) === true) {
+				$extensions[] = $key;
+			}
+
+			if ($isArray === false && $value === $mime) {
 				$extensions[] = $key;
 			}
```

Comment:
Since every if statement checks for array, I feel like this should be a larger wrapping if statement. Then we also wouldn't need `$isArray`.